### PR TITLE
Address duplicate name/ontology key issue

### LINKED ISSUE #20: Parse failure when `name` key is used in ontology header
The latest version of [OMRSE](https://github.com/ufbmi/OMRSE) includes both the `ontology` and `name` key in its header. This breaks the following lines:

https://github.com/dhimmel/obonet/blob/0ce7c81d7206ada2bff5204f3158847aa195dfe2/obonet/read.py#L29-L31

Minimal example:

```python
>>> import obonet
>>> url = 'https://github.com/ufbmi/OMRSE/raw/master/omrse-full.obo'
>>> graph = obonet.read_obo(url)
Traceback (most recent call last):
  File "/Users/cthoyt/dev/obonet/examples/ex.py", line 4, in <module>
    obonet.read_obo(url)
  File "/Users/cthoyt/dev/obonet/obonet/read.py", line 26, in read_obo
    graph = networkx.MultiDiGraph(
TypeError: networkx.classes.multidigraph.MultiDiGraph() got multiple values for keyword argument 'name'
```

Possible solutions:
- preemptively throw away `name` key from `headers`
- reassign `name` from `headers` to another key like in
   ```python
   if 'name' in headers:
      headers['long_name'] = headers.pop('name')
   ```

### PR DESCRIPTION
**Pull request by cthoyt**
Closes #20. I'm not sold on `long_name` as a key - it could be anything.

### CODE REVIEW

File: obonet/read.py [hunk 1]
```diff
@@ -26,6 +26,9 @@ def read_obo(path_or_file, ignore_obsolete=True):
     obo_file = open_read_file(path_or_file)
     typedefs, terms, instances, header = get_sections(obo_file)
     obo_file.close()
+    
+    if 'name' in header:
+        header['long_name'] = header.pop('name')
```
Comment (dhimmel) [inline obonet/read.py:31]:
```suggestion
        # OMRSE includes both the ontology and name key in its header
        # https://github.com/dhimmel/obonet/issues/20
        header['long_name'] = header.pop('name')
```

**Comment by dhimmel** [comment]
Ignoring backwards compatibility, do you think it'd be better to to just not remap `ontology` to `name`?

**Comment by cthoyt** [comment]
That would probably make things easier. Downstream, I rename the `name` to `prefix` and usually do some text processing to throw away some of the arbitrary information from ROBOT and other strange curation practices to make it match what's on identifiers.org

File: obonet/read.py (hunk 1 above)
Comment (cthoyt) [reply obonet/read.py]: Thanks for the suggestion but the way I updated it now is a bit more general and not exactly related to this issue

**Review by dhimmel — APPROVED** [review]
Thanks! Will merge and release.

**Comment by dhimmel** [comment]
Wow, I totally didn't notice the test was failing before merging. My bad. Looks like one of the tests is enforcing that the name key exists and is none when `ontology` is missing:

https://github.com/dhimmel/obonet/blob/c93e0fc7da34e4fe3871c5651aaf8610657273aa/tests/test_obo_reading.py#L59

This probably isn't critical functionality, so I can just change the test.

**Comment by dhimmel** [comment]
Released v0.3.0 https://pypi.org/project/obonet/0.3.0/ with this change and a bunch of packaging / modernization changes.

**Pull request merged**